spec/models/piece_spec.rb

```diff
@@ -4,13 +4,140 @@
+      white_pawn = FactoryGirl.create(:pawn, :white, game_id: game.id)
+
+      expect(white_pawn.capturable?(4, 5)).to eq false
+    end
+
+    it 'return false if specified coordinates occupied by same color piece' do
+      game = empty_game
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 0,
+        captured: false
+      )
+      other_white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(other_white_pawn.capturable?(
+               white_pawn.x_position,
+               white_pawn.y_position
+      )).to eq false
+    end
+
+    it 'return true if specified coordinates occupied by target piece' do
+      game = empty_game
+      black_rook = FactoryGirl.create(
+        :rook,
+        :black,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 0,
+        captured: false
+      )
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(white_pawn.capturable?(
+               black_rook.x_position,
+               black_rook.y_position
+      )).to eq true
+    end
+  end
+
+  describe 'capture!' do
+    it 'update target piece attributes' do
+      game = empty_game
+      black_rook = FactoryGirl.create(
+        :rook,
+        :black,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 1,
+        captured: false
+      )
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      black_rook.capture!(white_pawn.x_position, white_pawn.y_position)
+
+      white_pawn.reload
+
+      expect(white_pawn.x_position).to eq nil
+      expect(white_pawn.y_position).to eq nil
+      expect(white_pawn.captured).to eq true
+    end
+  end
+
+  describe 'friendly_piece?' do
+    it 'return true if target piece have the same color' do
+      game = empty_game
+      white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 3,
+        y_position: 0,
+        captured: false
+      )
+      other_white_pawn = FactoryGirl.create(
+        :pawn,
+        :white,
+        game_id: game.id,
+        x_position: 4,
+        y_position: 1,
+        captured: false
+      )
+
+      expect(other_white_pawn.friendly_piece?(
+               white_pawn.x_position,
+               white_pawn.y_position
+      )).to eq true
+    end
+  end
+
+  # before create a new game, destory all pieces pre-populated from game model
+  def empty_game
+    game = FactoryGirl.create(:game)
+    game.pieces.destroy_all
+    game.reload
+    game
+  end
```

Comment:
Using let or before seems to be the standard.   I don't think our app is complicated enough to worry about what is being mentioned in a blog post.  We are not experienced enough to be wading into what is obviously a controversial topic by the author of the blog themselves:  "For a while now, we’ve recommended against using let in RSpec and minitest. We even have a blog post explaining our thoughts. Despite that, I still have trouble explaining my problems with let and before to other developers. They’re wonderful for cleaning up duplication, and can make your tests smaller. That has to be a win, right?

+
Maybe not."

I think we should be focused on best practices and the easiest way to determine what a best practice is as a junior developer is to look at what the industry does.